backstory/src/main/java/com/google/sps/servlets/StoryManagerImpl.java

```diff
@@ -0,0 +1,167 @@
+    }
+    if (temperature < 0 || temperature > 1) {
+      this.temperature = 1.0;
+    }
+  }
+
+  /**
+   * Returns generation prefix.
+   *
+   * @return String The generation prefix.
+   */
+  public String getPrefix() {
+    return prefix;
+  }
+
+  /**
+   * Returns maximum length for generation.
+   *
+   * @return int The maximum length for text generation.
+   */
+  public int getMaxLength() {
+    return maxTextLength;
+  }
+
+  /**
+   * Returns temperature(volatility of generation).
+   *
+   * @return Double Numerical quantity representing temperature.
+   */
+  public Double getTemperature() {
+    return temperature;
+  }
+
+  /**
+   * Makes a post request with a JSON including GPT2 Parameters
+   *
+   * @returns HttpResponse The reponse from the Generation server expected to include
+   *          a "text" field with the generated text.
+   */
+  private HttpResponse makePostRequestGPT2() throws IOException {
+    // Form JSON body using generation parameters
+    String requestBody = makeRequestBody(prefix, maxTextLength, temperature);
+
+    // Build Request with Adapter and JSON Input
+    HttpRequest request = requestFactory.buildPostRequest(requestBody);
+    request.getHeaders().setContentType("application/json");
+
+    // Wait until response received
+    request.setConnectTimeout(0);
+    request.setReadTimeout(0);
+    return request.execute();
+  }
+
+  /**
+   * Returns generated text output using given fields.
+   *
+   * @return String Generated output text.
+   */
+  public String generateText() {
+    // Obtain response from Server POST Request
+    try {
+      HttpResponse outputResponse = makePostRequestGPT2();
+
+      // Parse response as JSON
+      try {
+        JSONObject jsonObject = new JSONObject(outputResponse.parseAsString());
+        return jsonObject.getString("text");
+      } catch (Exception jsonException) {
+        throw new RuntimeException("Failed to convert repsonse into JSON", jsonException);
+      }
+    } catch (IOException serverException) {
+      throw new RuntimeException("Error with server", serverException);
+    }
+  }
+
+  /**
+   * Allow public setting of RequestFactory for alternative posting.
+   *
+   * @param factory StoryManagerRequestFactory to use for HttpRequests.
+   */
+  public void setRequestFactory(StoryManagerRequestFactory factory) {
+    requestFactory = factory;
+  }
+
+  /**
+   * Forms request body string from GPT-2 parameters.
+   *
+   * @param prefix String to serve as generation prompt.
+   * @param maxLength Maximum text character length for generation output.
+   * @param temperature Double to hold number 0-1 for text generation volatility.
+   */
+  private String makeRequestBody(String prefix, int maxLength, Double temperature) {
+    String requestBody = "{\"length\": " + maxLength
+        + ",\"truncate\": \"<|endoftext|>\", \"prefix\": \"" + prefix
+        + "\", \"temperature\": " + temperature + "}";
+    return requestBody;
```

Comment:
I think you mentioned Gson not being able to handle HashMap<String, Object> to Json string, but it appears to work here (or am I confusing this with something else):
https://github.com/ykjchen/step/pull/2/files

Reply:
Yeah when I originally tested my strings were giving weird characters and StackOverflow mentioned this to be because of a <String, String> restriction; however, I just tried it end-to-end seems to work regardless when sent to the web app. I've implemented this now. Sorry for the confusion.